src/main/java/echoserver/EchoServerServiceRunnable.java

```diff
@@ -0,0 +1,36 @@
+package echoserver;
+
+import java.io.IOException;
+
+public class EchoServerServiceRunnable implements Runnable {
```

Comment:
Does `Runnable` need to be in the name here? It feels a little redundant. Especially in Java, I try to be conscious of this type of naming. It can lead to scenarios like this

```java
Runnable echoServerServiceRunnable = new EchoServerServiceRunnable();
```
vs something like

```java
Runnable service = new EchoServerService();
```